mlflow/tracking/client.py

```diff
@@ -910,6 +914,50 @@ def log_artifacts(self, run_id, local_dir, artifact_path=None):
         """
         self._tracking_client.log_artifacts(run_id, local_dir, artifact_path)
 
+    @contextlib.contextmanager
+    def _log_artifact_helper(self, run_id, artifact_file):
+        """
+        Yields a temporary path to store a file, and then calls `log_artifact` against that path.
+        """
+        norm_path = posixpath.normpath(artifact_file)
+        filename = posixpath.basename(norm_path)
+        artifact_path = posixpath.dirname(norm_path)
+        artifact_path = None if artifact_path == "" else artifact_path
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = os.path.join(tmp_dir, filename)
+            yield tmp_path
+            self.log_artifact(run_id, tmp_path, artifact_path)
+
+    def log_text(self, run_id, text, artifact_file):
+        """
+        Log a text as an artifact.
+
+        :param run_id: String ID of the run.
+        :param text: Text to log.
+        :param artifact_file: The run-relative artifact file path to which the text is saved.
```

Comment:
Should we allow a windows path (e.g. `"foo\\bar.txt"`) as `artifact_file `?

Reply:
`mlflow.log_artifact` doesn't accept a windows-style path:

```python
import mlflow


with mlflow.start_run() as run:
    mlflow.log_artifact("README.rst", "foo\\bar.txt")

    client = mlflow.tracking.MlflowClient()
    files = [a.path for a in client.list_artifacts(run.info.run_id)]
    print(files)
```

output:

```
['foo\\bar.txt']
```

Reply:
I don't think we need to support windows-style paths (technically the `artifact_file` is like a suffix to append to the run's artifact URI), but we should just make it clear in the `artifact_file` docstring that the specified path is expected to be in posixpath format

Reply:
Makes sense. I'll update the `artifact_file` description.

Reply:
Done in https://github.com/mlflow/mlflow/pull/3678/commits/b72d1c774d1dea7ebb0b227392b52ec4e586e169